code/modules/mob/living/simple_animal/friendly/diona.dm

```diff
@@ -92,6 +92,20 @@
 /mob/living/simple_animal/diona/UnarmedAttack(var/atom/A)
 	if(isdiona(A) && (src in A.contents)) //can't attack your gestalt
 		visible_message("[src] wiggles around a bit.")
+	// Nymphs can consume grown food as well as weeds.
+	else if(istype(A, /obj/item/reagent_containers/food/snacks/grown))
```

Comment:
This is a violation of OOP and is snowflake code. You want the behavior defined on the object in question. We already have `attack_animal` for stuff like this--you should be doing something like that, then only allowing for nymphs to benefit from this.

The idea is ok, but the code implementation is not.

Reply:
Unique interactions and type checks should be on the interacted, not the interactor (or a bunch of vaguely related objects) and that's what "snowflaked" means in most of the previous work I've done. That makes sense, thank you for your feedback.